### ISSUE TITLE
History folders are visible only after specific steps are performed 

### ISSUE THREAD

**Issue by alinamoldovan84** [mozilla-mobile/firefox-ios#14408]
## Steps to reproduce
1. Make sure you are logged with a Firefox account having some old data 
2. Open the hamburger menu and choose the history option
3. Check the history groups 

### Expected behavior
All history groups are visible 

### Actual behavior
Only Today and Yesterday groups are visible. 
Navigate outside history or open Recently closed section and come back -> Last week folder is displayed
Navigate outside history or open Recently closed section and come back  several times -> Last month folder is displayed

### Device & build information
* Device: iPhone 14 Plus (10.0.1), iPhone 13 Pro (15.7.1), iPad Air 3rd Gen (16.3.1)  
* Build version: 113.1 (30885

https://github.com/mozilla-mobile/firefox-ios/assets/124251579/717dff41-d8e4-4c62-869c-75c363571a41

) 
* First seen version: ? 
### Notes
Please note that this issue is reproducible on v113 (30644) too.
Attachments: 

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/FXIOS-6406)

**Comment by data-sync-user** [comment]
➤ Sorin Paraipan commented:

After investigating the behavior of the history feature, I discovered that it initially fetches 100 results. If more results are desired, pagination is available. While scrolling, an additional 100 results are fetched, and this process continues until there is no more history. I believe this implementation prioritizes performance by assuming that users do not need access to the entire history from the start. Instead, they can scroll to have access to a more past history if desired.
	To address the specific issue related to this ticket, when you open the history and then switch between tabs, returning to the history tab, it fetches another 100 results, this being a normal behavior.
	Here is a video demonstrating how history tab works:

!video.mp4|width=444,height=960!

I'd need Daniela Arcese's input regarding this, and if we need to change something.

**Comment by data-sync-user** [comment]
➤ Daniela Arcese commented:

Sorin Paraipan sure, can you please open the new ticket.

**Comment by data-sync-user** [comment]
➤ Sorin Paraipan commented:

Daniela Arcese Sure thing

**Comment by DianaBarladeanu** [comment]
> ➤ Sorin Paraipan commented:
> 
> Update: I fixed the bug in the sense that the sections will no longer load when switching between tabs. However, for the remaining issues, I suggest opening a new ticket to discuss how we can achieve the desired behavior. Many implementations will need to be changed. Does that sound okay to you, Daniela Arcese?

Verified as fixed on v9000 (32311) with iPhone SE (15.4).
